src/components/calcite-action/calcite-action.tsx

```diff
@@ -91,7 +91,29 @@ export class CalciteAction {
 
   @Element() el: HTMLCalciteActionElement;
 
-  private buttonEl: HTMLButtonElement;
+  @State() hasChildren = false;
+
+  buttonEl: HTMLButtonElement;
+
+  observer = new MutationObserver(() => this.setHasChildren());
```

Comment:
Would using `forceUpdate` after a mutation change help to simplify?

Reply:
I thought about that but wasn't sure if it was as efficient as using a state variable. Do you know?

Reply:
Haven't really played with it, but I'd think it's similar. Every time there's a mutation change you update state to rerender. Using `forceUpdate` seems like it would skip the setting state part. Worth testing.